scripts/tests/test_zkai_d64_stwo_vector_row_surface_probe.py

```
@@ -0,0 +1,135 @@
+            tsv_path = tmp / "probe.tsv"
+            PROBE.write_outputs(payload, json_path, tsv_path)
+
+            self.assertEqual(json.loads(json_path.read_text(encoding="utf-8"))["schema"], PROBE.SCHEMA)
+            tsv_lines = tsv_path.read_text(encoding="utf-8").splitlines()
+            self.assertEqual(tsv_lines[0].split("\t"), list(PROBE.TSV_COLUMNS))
+            self.assertIn("weight_commitment_consistency", "\n".join(tsv_lines))
+
+    def test_write_outputs_wraps_os_errors(self) -> None:
+        payload = PROBE.build_probe()
+        with mock.patch.object(pathlib.Path, "write_text", side_effect=OSError("disk full")):
+            with self.assertRaisesRegex(PROBE.D64VectorRowSurfaceError, "failed to write"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)